### PR TITLE
Cf/51 graphql api

### PR DESCRIPTION
**Pull request by cfranklin11**
- Add postgres database to the app
- Add django data models
- Add GraphQL schema, types, and route
- Add `predictions` query

### CODE REVIEW

File: server/tests/unit/test_schema.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    def test_predictions(self):
+        expected_predictions = [
+            {
+                'match': {'roundNumber': 5, 'year': 2018},
+                'mlModel': {'name': 'test_model'},
+                'isCorrect': True
+            },
+            {
+                'match': {'roundNumber': 7, 'year': 2014},
```
Comment (meligatt) [inline server/tests/unit/test_schema.py]: is this roundNumber camel case for some particular reason, or "just because and ain't gonna change it" ?

**Comment by meligatt** [comment]
@cfranklin11 I left a few comments..but oyu are gonna have to explain me the models for graphql part..I didnt get it at all XD

File: server/models.py [hunk 2]
```diff
@@ -1,3 +1,79 @@
...
+
+
+class TeamMatch(models.Model):
+    team = models.ForeignKey(Team, on_delete=models.CASCADE)
+    match = models.ForeignKey(Match, on_delete=models.CASCADE)
+    at_home = models.BooleanField()
+    score = models.PositiveSmallIntegerField()
+
+
+class MLModel(models.Model):
```
[the remark this answers is not in the document]
Comment (cfranklin11) [reply server/models.py]: `models` is the module that I'm importing. `Model` is a class in the `models` module and is the parent class of `MLModel`.

File: server/schema.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class PredictionType(DjangoObjectType):
+    class Meta:
+        model = Prediction
+
+    is_correct = graphene.Boolean()
+
+
+class Query(graphene.ObjectType):
```
[the remark this answers is not in the document]
Comment (cfranklin11) [reply server/schema.py]: see explanation above

File: server/tests/unit/models/test_prediction.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        self.home_team = Team(name='Richmond')
+        self.home_team.save()
+        self.away_team = Team(name='Melbourne')
+        self.away_team.save()
+        (TeamMatch(team=self.home_team, match=self.match, at_home=True, score=150)
+         .save())
+        (TeamMatch(team=self.away_team, match=self.match, at_home=False, score=100)
+         .save())
+
+    def test_is_correct(self):
```
[the remark this answers is not in the document]
Comment (cfranklin11) [reply server/tests/unit/models/test_prediction.py]: Naming convention for python test discovery: all test methods are named `test_<something>`. `is_correct` is the name of the method that I'm testing

File: server/tests/unit/test_schema.py (hunk 1 above)
Comment (cfranklin11) [reply server/tests/unit/test_schema.py]: GraphQL magically converts camelcase from the frontend to snakecase in the backend, then converts it back to camelcase in the JSON returned to the frontend. So, when matching attributes of the JSON returned by GraphQL, I need to use camelcase.

**Pull request merged**